File: manage-payments/python/change-amount.py
# The sandbox API key passed in 'authorization' is public.
# Don't submit any personally identifiable information in any requests made with this key.
# Sign in to developer.dojo.tech to create your own private sandbox key and use that instead
# for secure testing.

import http.client

conn = http.client.HTTPSConnection("api.dojo.tech")

payload = "{\"amount\":{\"value\":1000,\"currencyCode\":\"GBP\"}}"

headers = {
    'Content-Type': "application/json",
    'Version': "2022-04-07",
    'Authorization': "Basic sk_sandbox_<your_secret_key>"
    }

conn.request("POST", f"/payment-intents/{paymentIntentId}/amount", payload, headers)

res = conn.getresponse()
data = res.read()

print(data.decode("utf-8"))

conn.close()

File: manage-payments/php/change-amount.php
<?php
// The sandbox API key used in this example is public.
// Don't submit any personally identifiable information in any requests made with this key.
// Sign in to developer.dojo.tech to create your own private sandbox key and use that instead
// for secure testing.

namespace Test;
require_once "vendor/autoload.php";

use Dojo_PHP\ApiFactory;
use Dojo_PHP\Model\Money;
use Dojo_PHP\Model\SetAmountRequest;

$apiKey = "<your_api_key>";
$apiPaymentIntent = ApiFactory::createPaymentIntentApi($apiKey);

$customAmountRequest = new SetAmountRequest(["amount" => new Money(['value' => 1000, 'currency_code' => "GBP"])]);

$apiPaymentIntent->paymentIntentsSetCustomAmount(\Dojo_PHP\API_VERSION, "<paymentIntentId>", $customAmountRequest);